blog/03-15-2023-developing-a-crowdfunding-platform-for-social-causes-on-celo-blockchain-part-1/developing-a-crowdfunding-platform-for-social-causes-on-celo-blockchain-part-1.md

```diff
@@ -0,0 +1,378 @@
+
+- Function `myDonationsCount()`
+This function is used to get the count of donations made by the caller of the function.
+
+- Function `allDonationsCount()`
+This function is used to get the count of all donations made to the fundraiser.
+
+- Function `donate()`
+This function is used to make a donation to the fundraiser. It creates a Donation struct to store the value of the donation and the timestamp of the donation. It also creates an allDonation struct to store the address of the donor, the value of the donation, and the timestamp of the donation. The Donation struct is stored in the _donations mapping with the address of the donor as the key. The allDonation struct is stored in the _allDonations mapping with the ID of the fundraiser as the key. The value of the donation is added to the totalDonations variable, and the donationsCount and totalDonationsCount variables are incremented. Finally, an event DonationReceived is emitted with the address of the donor and the value of the donation as the parameters.
+
+- Function `myDonations()`
+This function is used to get the values and timestamps of all the donations made by the caller of the function.
+
+- Function `allDonations()`
+This function is used to get the addresses, values, and timestamps of all the donations made to the fundraiser.
+
+- Function `withdraw()`
+This function is used to withdraw the balance of the contract to the beneficiary. Only the owner of the contract can call this function.
+
+- Function `receive()`
+This function is called when ether is sent to the contract without a function call. It increments the totalDonations and donationsCount variables with the value of the donation.
+
+## Conclusion
+In this tutorial, we have developed a crowdfunding platform for social causes on the Celo blockchain. We used Truffle and React to build the smart contract and the front-end of the platform. We also deployed the smart contract to the Celo blockchain using Truffle.
+By using Celo, we can ensure that the crowdfunding platform is transparent, secure, and accessible to everyone, regardless of their location or financial status. We hope that this tutorial has been helpful in understanding how to develop decentralized applications on the Celo blockchain.
+
+## About Author
+
+Hi! My name is Kunal Dawar and I am a Full Stack web2/web3 Developer. I have participated in numerous hackathons and have been fortunate enough to win many of them.
+
+One thing that I am truly passionate about is creating things that are reliable and don't break easily. I believe that creating high-quality products is important not only for the users but also for the overall growth and success of a business.
+
+In my free time, I enjoy learning about new technologies and staying up-to-date with the latest trends in the field. I also love to share my knowledge with others and mentor those who are interested in pursuing a career in web development.
```

Comment:
Add next steps if there are any, or mention that part 2 is going to be there or something like that